quodlibet/ext/events/musicbrainzsync.py

```diff
@@ -0,0 +1,115 @@
+# Copyright 2019 LoveIsGrief
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+from quodlibet import plugins, qltk
+from quodlibet.qltk.entry import UndoEntry
+from quodlibet.util.thread import call_async_background, Cancellable
+
+try:
+    import musicbrainzngs
+except ImportError:
+    raise plugins.MissingModulePluginException("musicbrainzngs")
+
+from gi.repository import Gtk
+from quodlibet import _
+from quodlibet.plugins import PluginConfig
+from quodlibet.plugins.events import EventPlugin
+
+ATTR_BRAINZ = 'musicbrainz_trackid'
+ATTR_RATING = '~#rating'
+
+BRAINZ_APP = "quodlibetMusicBrainzSync"
+VERSION = "0.1"
+
+plugin_config = PluginConfig("musicbrainz-sync")
+defaults = plugin_config.defaults
+defaults.set("username", "")
+defaults.set("password", "")
+
+
+class MusicBrainzSyncPlugin(EventPlugin):
+    PLUGIN_ID = "musicbrainzsync"
+    VERSION = VERSION
+    PLUGIN_NAME = _("MusicBrainz Sync")
+    PLUGIN_DESC = _("Syncs the rating of a song with music brainz.")
+
+    def __init__(self):
+        super(MusicBrainzSyncPlugin, self).__init__()
+        musicbrainzngs.set_rate_limit()
+        musicbrainzngs.set_useragent(
+            BRAINZ_APP, VERSION, "loveisgrief@tuta.io"
```

Comment:
Might that want to be the user's email, or the default QL one (we use this for ratings actually)

Reply:
Opted for the default QL on.